tests/Functional/Platform/DateExpressionTest.php

```diff
@@ -30,19 +31,21 @@ public function testDifference(string $date1, string $date2, int $expected): voi
         self::assertEquals($expected, $diff);
     }
 
-    /** @return string[][]|int[][] */
+    /** @return array<string, array{string, string, int}> */
     public static function differenceProvider(): iterable
     {
-        $date1    = new DateTimeImmutable();
-        $date2    = new DateTimeImmutable('2018-04-10 10:10:10');
-        $expected = $date1->modify('midnight')->diff(
-            $date2->modify('midnight'),
-        )->days;
+        $datetime1 = new DateTimeImmutable();
+        $datetime2 = new DateTimeImmutable('2018-04-10 10:10:10');
+        $date1     = $datetime1->modify('midnight');
+        $date2     = $datetime2->modify('midnight');
+        assert($date1 !== false && $date2 !== false);
+        $expected = $date1->diff($date2)->days;
```

Comment:
Not exactly related but if the expected value is computed programmatically, the test is not isolated. I'd remove this "dynamic" case instead of fixing it.

Reply:
You're right, plus it's way too complex.